plugins/c8y_configuration_plugin/src/main.rs

```diff
@@ -94,22 +101,27 @@ async fn main() -> Result<(), anyhow::Error> {
     let config_repository = tedge_config::TEdgeConfigRepository::new(tedge_config_location.clone());
     let tedge_config = config_repository.load()?;
 
-    let plugin_config = PluginConfig::new(config_plugin_opt.config_file);
-
-    // Create required clients
+    let plugin_config = PluginConfig::new(&config_plugin_opt.config_file);
     let mqtt_port = tedge_config.query(MqttPortSetting)?.into();
     let mut http_client = create_http_client(&tedge_config).await?;
-
     let tmp_dir = tedge_config.query(TmpPathSetting)?.into();
 
-    run(plugin_config, mqtt_port, &mut http_client, tmp_dir).await
+    run(
+        plugin_config,
+        mqtt_port,
+        &mut http_client,
+        tmp_dir,
+        &config_plugin_opt.config_file,
+    )
+    .await
 }
 
 async fn run(
     plugin_config: PluginConfig,
     mqtt_port: u16,
     http_client: &mut impl C8YHttpProxy,
     tmp_dir: PathBuf,
+    config_file_path: &Path,
```

Comment:
Why add this argument when `plugin_config` is already there? Seems like a duplicate. I think you can replace the existing `plugin_config: PluginConfig` argument with this one and always create the `PluginConfig` object inside this function.